routes/donate.js

```
@@ -1,133 +1,280 @@
+                    db("subscriptions")
+                    .where({
+                        customer_id: sub.customer_id
+                    })
+                    .del()
+                    .asCallback(cb);
+                })
+            }, function(err) {
+                if (err) return next(err);
+                
+                req.session.cancel = true;
+                res.redirect("/thanks");
+            })
+        })
     });
-    return donate;
+    donate.route("/stripe_endpoint").post(function(req, res, next) {
+        var id = req.body.id;
+        console.log("Got a event from Stripe, id %s", id);
+        // Get the event from Stripe to verify
+        stripe.events.retrieve(id, function(err, event) {
+            if (err) return res.sendStatus(400);
+            
+            // Only care about charge succeeded or subscription ended
+            if (event.type !== "charge.succeeded" && event.type !== "customer.subscription.deleted") {
+                return res.sendStatus(200);
+            }
+            
+            // Check that we haven't seen this before
+            redis.lrange("stripe:events", 0, 1000, function(err, result) {
+                if (err) return res.sendStatus(400); // Redis is derping, have Stripe send back later
 
+                for (var e in result) {
+                    
+                    if (e == id) {
+                        console.log("Found event %s in redis.", id);
+                        return res.sendStatus(200);
+                    }
+                }
+                
+                // New event
+                if (event.type === "charge.succeeded") {
+                    var amount = event.data.object.amount/100;
+                    
+                    console.log("Event %s: Charge succeeded for %s.", id, amount);
+                    
+                    // Update cheese goal
+                    redis.incrby("cheese_goal", amount, function(err, val) {
```

do you just want to swallow redis exception?  at least log it